File: src/test/java/com/raychenon/leetcode/SearchInsertPositionTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 2019-07-15
 * https://leetcode.com/problems/search-insert-position/
 */
public class SearchInsertPositionTest {

    private static SearchInsertPosition searchInsertPosition;

    @BeforeClass
    public static void setUp() {
        searchInsertPosition = new SearchInsertPosition();
    }

    @Test
    public void testSearchInsert_WhenTargetIsInArray() {
        int sortedArray[] = {1, 3, 5, 6};
        int position = searchInsertPosition.searchInsert(sortedArray, 5);
        assertEquals(2, position);
    }

    @Test
    public void testSearchInsert_WhenTargetNotInArray() {
        int sortedArray[] = {1, 3, 5, 6};
        int position = searchInsertPosition.searchInsert(sortedArray, 2);
        assertEquals(1, position);
    }

    @Test
    public void testSearchInsert_WhenTargetIsBiggerThanArrayMax() {
        int sortedArray[] = {1, 3, 5, 6};
        int position = searchInsertPosition.searchInsert(sortedArray, 7);
        assertEquals(4, position);
    }

    @Test
    public void testSearchInsert_WhenTargetIsSmallerThanArrayMin() {
        int sortedArray[] = {1, 3, 5, 6};
        int position = searchInsertPosition.searchInsert(sortedArray, 0);
        assertEquals(0, position);
    }
}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/SearchInsertPositionTest.kt
package com.raychenon.kotlin.leetcode

import org.junit.Assert
import org.junit.Test

/**
 * User: raychenon
 * Date: 2019-07-13
 * https://leetcode.com/problems/search-insert-position/submissions/
 */
class SearchInsertPositionTest {

    @Test
    fun testSearchInsert_WhenInArrayIsEmpty() {
        val position = SearchInsertPosition.searchInsert(intArrayOf(), 5)
        Assert.assertEquals(0, position)
    }

    @Test
    fun testSearchInsert_WhenTargetIsInArray() {
        val position = intArrayOf(1, 3, 5, 6).searchInsert(5)
        Assert.assertEquals(2, position)
    }

    @Test
    fun testSearchInsert_WhenTargetNotInArray() {
        val position = SearchInsertPosition.searchInsert(intArrayOf(1, 3, 5, 6), 2)
        Assert.assertEquals(1, position)
    }

    @Test
    fun testSearchInsert_WhenTargetIsBiggerThanArrayMax() {
        val position = SearchInsertPosition.searchInsert(intArrayOf(1, 3, 5, 6), 7)
        Assert.assertEquals(4, position)
    }

    @Test
    fun testSearchInsert_WhenTargetIsSmallerThanArrayMin() {
        val position = SearchInsertPosition.searchInsert(intArrayOf(1, 3, 5, 6), 0)
        Assert.assertEquals(0, position)
    }

}